### PR TITLE
B022: No arguments passed to contextlib.suppress

### LINKED ISSUE #222: Detect when no arguments were passed to 'contextlib.suppress'
Wrong code that is not detected by Flake8:

```python
import contextlib

with contextlib.suppress():  
    raise ValueError
```
Where the Flake8 error must be raised:

```python
import contextlib

with contextlib.suppress():  # <-- ERROR: NO ARGUMENTS WERE PASSED!
    raise ValueError
```

How the correct code may look:

```python
import contextlib

with contextlib.suppress(ValueError):  
    raise ValueError
```

### PR DESCRIPTION
**Pull request by jpy-git**
Closes #222. 

This covers the case where `contextlib.suppress` is used
```
import contextlib

with contextlib.suppress():  
    raise ValueError
```

However, it does not currently cover the case where `suppress` is imported from `contextlib`
```
from contextlib import suppress

with suppress():  
    raise ValueError
```
Do you know of any existing flake8 helper functions/methods to tie the `suppress` function node to it's import? 
I don't want to just ignore the one word `suppress` without checking incase it's actually a user defined function.

### CODE REVIEW

**Comment by cooperlees** [comment]
You're to quick - Sorry - I just merged another PR. I can look at cleaning up here or if you beat me that would be awesome too :D

**Comment by jpy-git** [comment]
> You're to quick - Sorry - I just merged another PR. I can look at cleaning up here or if you beat me that would be awesome too :D

hehe no worries, I've resolved them now 😄

Would be good to get your take on this PR currently only covering `contextlib.suppress` rather just `suppress` since we'd have to resolve scope to figure out that the `suppress` being called is definitely the one from `contextlib`?

**Review by cooperlees — APPROVED** [review]
All makes sense to me, and test show us working and handling `*` expansion which is good.

**Comment by cooperlees** [comment]
> Would be good to get your take on this PR currently only covering `contextlib.suppress` rather just `suppress` since we'd have to resolve scope to figure out that the `suppress` being called is definitely the one from `contextlib`?

Ahhh, I didn't think of this. Yeah, I feel assuming with `suppress` would be to noisy and we shouldn't do that.

File: README.rst [hunk 1]
```diff
@@ -141,6 +141,10 @@ garbage collection.
 **B021**: f-string used as docstring. This will be interpreted by python
 as a joined string rather than a docstring.
 
+**B022**: No arguments passed to `contextlib.suppress`.
+No exceptions will be suppressed and therefore this context manager is redundant.
```
Comment (cooperlees) [inline README.rst:145]: Maybe we could note here our miss of `from contextlib import suppress` and the reason why - noise as we're sure other people call methods suppress ...

**Comment by jpy-git** [comment]
> Ahhh, I didn't think of this. Yeah, I feel assuming with `suppress` would be to noisy and we shouldn't do that.

Yeah that was my thinking too, better to avoid false negatives 😄 

Maybe we initially keep the rule like this and then I can make a separate issue to find a way to track variable/function scope when traversing the AST as a future refinement (I think pyflakes has something similar for tracking if imports get used).

File: README.rst (hunk 1 above)
Comment (jpy-git) [reply README.rst:145]: Good shout, will add a note 👍

Comment (jpy-git) [reply README.rst:145]: I've added that in now

**Pull request merged**